bitcoin/src/bip152.rs

```diff
@@ -158,7 +158,32 @@ pub struct HeaderAndShortIds {
     ///  which we expect a peer may be missing.
     pub prefilled_txs: Vec<PrefilledTransaction>,
 }
-impl_consensus_encoding!(HeaderAndShortIds, header, nonce, short_ids, prefilled_txs);
+
+impl Decodable for HeaderAndShortIds {
+    fn consensus_decode<R: BufRead + ?Sized>(r: &mut R) -> Result<Self, encode::Error> {
+        let header_short_ids = HeaderAndShortIds {
+            header: Decodable::consensus_decode(r)?,
+            nonce: Decodable::consensus_decode(r)?,
+            short_ids: Decodable::consensus_decode(r)?,
+            prefilled_txs: Decodable::consensus_decode(r)?
+        };
+        if header_short_ids.short_ids.len() + header_short_ids.prefilled_txs.len() > u16::MAX.into() {
```

Comment:
The `+` operation could overflow though. :D But I think it's OK, it'd have to fill whole RAM to do that so it'd be just OOM killed. (And we have the protection against this on another layer.)